src/TestFramework/TestFramework/Assertions/Assert.Items.cs

```diff
@@ -0,0 +1,350 @@
+        // the BCL's StringBuilder.AppendInterpolatedStringHandler
+        public void AppendFormatted<T>(T value, string? format)
+            => _builder!.AppendFormat(null, $"{{0:{format}}}", value);
+
+        public void AppendFormatted<T>(T value, int alignment)
+            => _builder!.AppendFormat(null, $"{{0,{alignment}}}", value);
+
+        public void AppendFormatted<T>(T value, int alignment, string? format)
+            => _builder!.AppendFormat(null, $"{{0,{alignment}:{format}}}", value);
+
+        public void AppendFormatted(string? value)
+            => _builder!.Append(value);
+
+#pragma warning disable RS0026 // Do not add multiple public overloads with optional parameters
+#pragma warning disable RS0027 // API with optional parameter(s) should have the most parameters amongst its public overloads
+        public void AppendFormatted(string? value, int alignment = 0, string? format = null)
+            => _builder!.AppendFormat(null, $"{{0,{alignment}:{format}}}", value);
+
+        public void AppendFormatted(object? value, int alignment = 0, string? format = null)
+            => _builder!.AppendFormat(null, $"{{0,{alignment}:{format}}}", value);
+#pragma warning restore RS0026 // Do not add multiple public overloads with optional parameters
+#pragma warning restore RS0027 // API with optional parameter(s) should have the most parameters amongst its public overloads
+    }
+
+    public static void IsNotEmpty<T>(IEnumerable<T> collection)
+        => IsNotEmpty(collection, string.Empty, null);
+
+    public static void IsNotEmpty<T>(IEnumerable<T> collection, string? message)
+        => IsNotEmpty(collection, message, null);
+
+#pragma warning disable IDE0060 // Remove unused parameter
+    public static void IsNotEmpty<T>(IEnumerable<T> collection, [InterpolatedStringHandlerArgument(nameof(collection))] ref AssertAnyInterpolatedStringHandler<T> message)
+#pragma warning restore IDE0060 // Remove unused parameter
+        => message.ComputeAssertion();
+
+    public static void IsNotEmpty<T>(IEnumerable<T> collection, [StringSyntax(StringSyntaxAttribute.CompositeFormat)] string? message, params object?[]? parameters)
+    {
+        if (collection.Any())
+        {
+            return;
+        }
+
+        string userMessage = BuildUserMessage(message, parameters);
+        ThrowAssertIsNotEmptyFailed(userMessage);
+    }
+
+    public static void HasCount<T>(int expected, IEnumerable<T> collection)
+        => HasCount(expected, collection, string.Empty, null);
+
+    public static void HasCount<T>(int expected, IEnumerable<T> collection, string? message)
+        => HasCount(expected, collection, message, null);
+
+#pragma warning disable IDE0060 // Remove unused parameter
+    public static void HasCount<T>(int expected, IEnumerable<T> collection, [InterpolatedStringHandlerArgument(nameof(expected), nameof(collection))] ref AssertCountInterpolatedStringHandler<T> message)
```

Comment:
Should `AssertCountInterpolatedStringHandler` be renamed to `AssertHasCountInterpolatedStringHandler`?